Code/Tools/AssetProcessor/native/InternalBuilders/SettingsRegistryBuilder.cpp

```diff
@@ -279,18 +285,47 @@ namespace AssetProcessor
 
                     for (const auto& settingsKey : settingsToCopy)
                     {
-                        AZ::SettingsRegistryInterface::FixedValueString settingsValue;
+                        FixedValueString settingsValue;
                         [[maybe_unused]] bool settingsCopied = settingsRegistry->Get(settingsValue, settingsKey)
                             && registry.Set(settingsKey, settingsValue);
                         AZ_Warning("Settings Registry Builder", settingsCopied, "Unable to copy setting %s from AssetProcessor settings registry"
                             " to local settings registry", settingsKey.c_str());
                     }
+
+                    // Read the AssetProcessor loaded Gem Information from the global Registry
+                    AZStd::vector<AzFramework::GemInfo> gemInfos;
+                    size_t pathIndex{};
+                    if (AzFramework::GetGemsInfo(gemInfos, *settingsRegistry))
+                    {
+                        AZStd::vector<AZ::IO::PathView> sourcePaths;
+                        for (const AzFramework::GemInfo& gemInfo : gemInfos)
+                        {
+                            for (const AZ::IO::Path& absoluteSourcePath : gemInfo.m_absoluteSourcePaths)
+                            {
+                                if (auto foundIt = AZStd::find(sourcePaths.begin(), sourcePaths.end(), absoluteSourcePath);
+                                    foundIt == sourcePaths.end())
+                                {
+                                    sourcePaths.emplace_back(absoluteSourcePath);
+                                }
+                            }
+                        }
+
+                        for (const AZ::IO::Path& sourcePath : sourcePaths)
+                        {
+                            // Use JSON Pointer to append elements to the SourcePaths array
+                            registry.Set(FixedValueString::format("%s/SourcePaths/%zu", PlaceholderGemKey.c_str(), pathIndex++),
+                                sourcePath.Native());
+                        }
+                    }
                 }
 
                 AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_EngineRegistry(registry, platform, specialization, &scratchBuffer);
                 AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_GemRegistries(registry, platform, specialization, &scratchBuffer);
                 AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ProjectRegistry(registry, platform, specialization, &scratchBuffer);
 
+                // The Placeholder Key is removed now that the each gem "<gem-source-path>/Registry" directory has been merged
```

Comment:
Typo: "that the each" -> "that each".

Reply:
I feel like the new chunk of code here actually does need a comment with an explanation why this is required (ie, order of events causes us to need to do this)